src/chain.rs

```diff
@@ -15,109 +16,60 @@ lazy_static! {
 }
 
 #[derive(Debug, Copy, Clone, PartialEq, Hash, Serialize, Ord, PartialOrd, Eq)]
-pub enum Network {
+pub struct Network {
+    pub network_type: NetworkType,
```

Comment:
How abount only `type` instead of `network_type`?

Reply:
`type` is a reserved word, so using `type` generates a build error.

Reply:
I see.